frontend/src/main/scala/bloop/bsp/ScalaTestSelection.scala

```diff
@@ -0,0 +1,31 @@
+package bloop.bsp
+
+import io.circe.derivation.JsonCodec
+import io.circe.derivation.deriveDecoder
+import io.circe.derivation.deriveEncoder
+import io.circe.Decoder
+import io.circe.Encoder
+
+/**
+  * Below datatypes are based on https://github.com/build-server-protocol/build-server-protocol/issues/249#issuecomment-983435766
+  */
+
+case class ScalaTestSelection(
+    jvmOptions: List[String],
+    classes: List[ScalaTestSuiteSelection],
+    env: Map[String, String],
+)
+
+object ScalaTestSelection {
+  implicit val decoder: Decoder[ScalaTestSelection] = deriveDecoder
+  implicit val encoder: Encoder[ScalaTestSelection] = deriveEncoder
+}
+
+case class ScalaTestSuiteSelection(
+    className: String,
```

Comment:
I don't see a reason why to limit ourselves only to a single class per run. Currently, this might not be needed, but it's trivial to use current implementation to run single files, the opposite on the contrary won't be possible and taking into account SDAP overhead also won't be efficient.

Reply:
> I don't see a reason why to limit ourselves only to a single class per run

Sure, but do we need it? I would say no, we can run them separately and I don't estimate there will be a sensible UI that will allow you to choose test classes to run. Even if there is, we can just run them concurrently in separate sessions.

Reply:
According to what we agreed on slack, I'll leave this for now as it currently is ;)